packages/react-aria-components/test/GridList.test.js

```diff
@@ -472,6 +472,55 @@ describe('GridList', () => {
     expect(document.activeElement).toBe(document.body);
   });
 
+  it('should support rendering a TagGroup with tabbing navigation inside a GridListItem', async () => {
+    let buttonRef = React.createRef();
+    let onRemove = jest.fn();
+    let {getAllByRole} = render(
+      <GridList aria-label="Test" keyboardNavigationBehavior="tab">
+        <GridListItem data-test-id="grid-list" id="tags" textValue="tags">
+          <TagGroup aria-label="Tag group" onRemove={onRemove}>
+            <TagList>
+              <Tag id="1" textValue="Tag 1">Tag 1<Button slot="remove">X</Button></Tag>
+              <Tag id="2" textValue="Tag 2">Tag 2<Button slot="remove">X</Button></Tag>
+              <Tag id="3" textValue="Tag 3">Tag 3<Button slot="remove">X</Button></Tag>
+            </TagList>
+          </TagGroup>
+        </GridListItem>
+        <GridListItem id="dog" textValue="Dog">Dog <Button aria-label="Info" ref={buttonRef}>ⓘ</Button></GridListItem>
+        <GridListItem id="kangaroo">Kangaroo</GridListItem>
+      </GridList>
+    );
+
+    let items = getAllByRole('grid')[0].children;
+    let tags = within(items[0]).getAllByRole('row');
+
+    await user.tab();
+    expect(document.activeElement).toBe(items[0]);
+
+    await user.keyboard('{ArrowRight}');
+    expect(document.activeElement).toBe(items[0]);
+
+    await user.tab();
+    expect(document.activeElement).toBe(tags[0]);
+
+    await user.keyboard('{ArrowRight}');
+    expect(document.activeElement).toBe(tags[1]);
+
+    await user.tab();
+    expect(document.activeElement).toBe(within(tags[1]).getByRole('button'));
+
+    await user.keyboard(' ');
+    expect(onRemove).toHaveBeenCalledTimes(1);
+    expect(onRemove).toHaveBeenLastCalledWith(new Set(['2']));
+
+    await user.tab({shift: true});
+    expect(document.activeElement).toBe(tags[1]);
+
+    // TODO: Fix, this is going to the previous tag's remove button instead of to the gridlist item
```

Comment:
Found a test case that's broken, however, I don't think it's due to any of your work, I think this just surfaced it. If you want to have a look go ahead.